source/extensions/filters/network/rocketmq_proxy/app_load_balancer_impl.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <vector>
+
+#include "extensions/filters/network/rocketmq_proxy/app_load_balancer.h"
+#include "extensions/filters/network/rocketmq_proxy/conn_manager.h"
+#include "extensions/filters/network/rocketmq_proxy/message_queue.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+
+class MessageQueue;
+
+/**
+ * This class implements message consuming load balancing according to annotated version and
+ * dynamically configured version-quota. In other words, operators may arbitrarily change percent of
+ * messages allocated to a certain group of consumers to consume in quasi real-time.
+ */
+class VersionBasedAppLoadBalancer : public AppLoadBalancer {
+public:
+  explicit VersionBasedAppLoadBalancer(ConnectionManager& connectionManager)
+      : connection_manager_(connectionManager), active_(false) {}
+
+  std::string lb(RemotingCommandPtr& cmd) override;
+
+  void update(absl::string_view cluster_name) override;
+
+  void add(const std::string& key, const std::vector<MessageQueue>& queues) override;
+
+protected:
+  bool active() override;
+
+private:
+  ConnectionManager& connection_manager_;
+  bool active_;
+  absl::flat_hash_map<std::string /*group@topic*/, uint32_t> index_map_;
+  absl::flat_hash_map<std::string /*group@topic*/, std::vector<MessageQueue>> queue_map_;
+
+  const static std::string VERSION_QUOTA_KEY;
```

Comment:
Where are they used and defined?

Reply:
OK, this `VERSION_QUOTA_KEY` has been deprecated, I would delete it.